File: rust/rustbook/ch1/python/prime100.py
# 素数を100個生成する


def is_prime(n):
    # range 2〜nまで繰り替えす
    for i in range(2, n):
        if n % i == 0:
            return False
    return True


# count個の素数を生成する


def get_primes(count):
    # リスト型
    res = []
    i = 2
    # len は要素の数を取得
    while len(res) < count:
        if is_prime(i):
            res.append(i)
        i += 1
    return res


print(get_primes(100))

File: rust/rustbook/ch1/rust/src/prime100.rs
//素数を100個生成する

fn is_prime(n: usize) -> bool {
    for i in 2..n {
        if n % i == 0 {
            return false;
        }
    }
    return true;
}
//素数100個を求める
fn get_primes(primes: &mut [usize; 100]) {
    let mut i = 2;
    let mut count = 0;

    while count < 100 {
        if is_prime(i) {
            primes[count] = i;
            count += 1;
        }
        i += 1;
    }
}

fn main() {
    //初期値0の配列を100個用意する
    let mut primes = [0; 100];
    get_primes(&mut primes);

    println!("{:?}",primes);
}
